cpp/include/cudf/types.h

```diff
@@ -177,7 +177,28 @@ typedef enum {
   N_GDF_AGG_OPS,      ///< The total number of aggregation operations. ALL NEW OPERATIONS SHOULD BE ADDED ABOVE THIS LINE
 } gdf_agg_op;
 
+/**
+ * @brief These enums indicate the supported reduction operations that can be
+ * performed on a column
+ */
+typedef enum {
+  GDF_REDUCTION_SUM = 0,        ///< Computes the sum of all values in the column
+  GDF_REDUCTION_MIN,            ///< Computes the minimum of all values in the column
+  GDF_REDUCTION_MAX,            ///< Computes the maximum of all values in the column
+  GDF_REDUCTION_PRODUCT,        ///< Computes the multiplicative product of all values in the column
+  GDF_REDUCTION_SUMOFSQUARES,   ///< Computes the sum of squares of the values in the column
+} gdf_reduction_op;
 
+/**
+ * @brief These enums indicate the supported operations of prefix scan that can be
+ * performed on a column
+ */
+typedef enum {
+  GDF_SCAN_SUM = 0,             ///< Computes the prefix scan of sum operation of all values for the column
+  GDF_SCAN_MIN,                 ///< Computes the prefix scan of maximum operation of all values for the column
+  GDF_SCAN_MAX,                 ///< Computes the prefix scan of maximum operation of all values for the column
+  GDF_SCAN_PRODUCT,             ///< Computes the prefix scan of multiplicative product operation of all values for the column
+} gdf_scan_op;
```

Comment:
@jrhemstad I was hoping we could consolidate all these duplicated operator enums somehow and have a single enum type for operators across all algorithms that operate or aggregate. I realize that some operators are unique to aggregations, but, for example SCAN_SUM, SCAN_PRODUCT, SCAN_MIN, SCAN_MAX are the same as REDUCE_SUM, REDUCE_PRODUCT, REDUCE_MIN, REDUCE_MAX, which are the same as OP_ADD, OP_MULTIPLY, OP_MIN, OP_MAX, so there is a lot of duplication we could avoid.

Reply:
I'm absolutely in favor of consolidating the enums (we could use for groupby and rolling windows as well), my only concern is that if we consolidate down to a single list of enums that it will be potentially confusing if some features only support a subset of those operators. E.g., `cudf::reduction` supports a `product` reduction, but `groupby` does not, etc. 

By having distinct enums for each function, it's more explicit about what is supported/unsupported.

Perhaps in time all functions that accept a consolidated `aggregation_operation` enum will support all of the listed operations and this won't be a problem.